client/state/data-layer/wpcom/read/lists/feeds/new/index.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice } from 'state/notices/actions';
+import { READER_LIST_ITEM_ADD_FEED } from 'state/reader/action-types';
+import { receiveReaderListAddFeed } from 'state/reader/lists/actions';
+import { registerHandlers } from 'state/data-layer/handler-registry';
+
+registerHandlers( 'state/data-layer/wpcom/read/lists/feeds/new/index.js', {
+	[ READER_LIST_ITEM_ADD_FEED ]: [
+		dispatchRequest( {
+			fetch: ( action ) =>
+				http(
+					{
+						method: 'POST',
+						path: `/read/lists/${ action.listOwner }/${ action.listSlug }/feeds/new`,
+						apiVersion: '1.2',
+						body: {
+							// Only one of these will be set
+							feed_url: action.feedUrl,
+							site_id: action.siteId,
+							feed_id: action.feedId,
+						},
+					},
+					action
+				),
+			onSuccess: ( action, apiResponse ) =>
+				receiveReaderListAddFeed( action.listOwner, action.listSlug, apiResponse.feed_id ),
```

Comment:
since we're now loading feed data via meta ... we'll need to add in some backup to ask for the feed info one off maybe?